src/blocks/keyboard_layout.rs

```diff
@@ -153,6 +155,33 @@ struct Info {
     variant: Option<String>,
 }
 
+struct XkbSwitch;
+
+impl XkbSwitch {
+    fn new() -> Self {
+        Self
+    }
+}
+
+#[async_trait]
+impl Backend for XkbSwitch {
+    async fn get_info(&mut self) -> Result<Info> {
+        let output = Command::new("xkbswitch").arg("-p").output().await?;
+        let output = String::from_utf8_lossy(&output.stdout);
+
+        let mut parts = output.split(':');
+        let layout = parts.next().unwrap_or_default().trim().to_string();
+        let variant = parts.next().map(|v| v.trim().to_string());
+
+        Ok(Info { layout, variant })
+    }
+
+    async fn wait_for_change(&mut self) -> Result<()> {
+        // This backend doesn't support waiting for changes
+        Ok(())
```

Comment:
Then it should `sleep()` (see `SetXkbMap`)

Reply:
thanks

Reply:
I have made the changes that you told me about. Are there any other changes I should make to make this working